# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.h
# Commit: 807f1d1ada66
# Full Hash: 807f1d1ada66856a25544239f6955f676a1d011e
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 4: Implement ActorDestroy r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227689
# ==============================================================================

diff -r 22abfa77d4fc -r 807f1d1ada66 dom/notification/NotificationParent.h
--- a/dom/notification/NotificationParent.h	Mon Nov 11 20:34:12 2024 +0000
+++ b/dom/notification/NotificationParent.h	Mon Nov 11 20:34:12 2024 +0000
@@ -35,21 +35,32 @@
         mOptions(aOptions) {};
 
   IPCResult RecvShow(ShowResolver&& aResolver);
-  IPCResult RecvClose();
 
   nsresult BindToMainThread(
       Endpoint<PNotificationParent>&& aParentEndpoint,
       PBackgroundParent::CreateNotificationParentResolver&& aResolver);
 
+  void ActorDestroy(ActorDestroyReason aWhy) override;
+
  private:
   ~NotificationParent() = default;
 
+  void GetAlertName(nsAString& aRetval) {
+    if (mAlertName.IsEmpty()) {
+      MaybeInitAlertName();
+    }
+    aRetval = mAlertName;
+  }
+  void MaybeInitAlertName();
+
   NotNull<nsCOMPtr<nsIPrincipal>> mPrincipal;
   NotNull<nsCOMPtr<nsIPrincipal>> mEffectiveStoragePrincipal;
   bool mIsSecureContext;
   nsString mId;
   nsString mScope;
   IPCNotificationOptions mOptions;
+
+  nsString mAlertName;
 };
 
 }  // namespace mozilla::dom::notification
